File: 145. Binary Tree Postorder Traversal.js
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function(root) {

    const result = [];

    function postorder(node) {

        // Base case
        if (!node) {
            return;
        }

        // Left
        postorder(node.left);

        // Right
        postorder(node.right);

        // Root
        result.push(node.val);
    }

    postorder(root);

    return result;
};

File: 145. Binary Tree Postorder Traversal.java
class Solution {

    private List<Integer> result = new ArrayList<>();

    private void postorder(TreeNode root) {

        // Base case
        if (root == null) {
            return;
        }

        // Traverse left subtree
        postorder(root.left);

        // Traverse right subtree
        postorder(root.right);

        // Visit current node
        result.add(root.val);
    }

    public List<Integer> postorderTraversal(TreeNode root) {

        postorder(root);

        return result;
    }
}
